packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ProjectController.java

```diff
@@ -1208,32 +1211,80 @@ public ResponseEntity<JsonNode> makeProjectSample(@PathVariable("id") final UUID
 				throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("projects.not-found"));
 			}
 
-			// Update the project and make it public as well as a sample project
-			project.get().setSampleProject(true).setPublicAsset(true);
-			projectAssetService.togglePublicForAssets(terariumAssetServices, id, true, Schema.Permission.WRITE);
-			projectService.updateProject(project.get());
+			// If the user is making the project a sample
+			if (isSample) {
+				// Check if the project is already a sample project
+				if (project.get().getSampleProject()) {
+					throw new ResponseStatusException(HttpStatus.BAD_REQUEST, messages.get("projects.already-sample"));
+				}
 
-			/* Permissions */
+				// Update the project and make it public as well as a sample project
+				project.get().setSampleProject(true).setPublicAsset(true);
+				projectAssetService.togglePublicForAssets(terariumAssetServices, id, true, Schema.Permission.WRITE);
+			} else {
+				// Check if the project is not a sample project
+				if (!project.get().getSampleProject()) {
+					throw new ResponseStatusException(HttpStatus.BAD_REQUEST, messages.get("projects.not-sample"));
+				}
+
+				// Update the project, but do not change the public status
+				project.get().setSampleProject(false);
+			}
 
-			// Getting the project
+			// Update the project
+			projectService.updateProject(project.get());
+
+			/* Project Permissions */
 			final RebacProject rebacProject = new RebacProject(id, reBACService);
 
-			// Delete all previous user relationships attached to the project,
+			// Update the user permissions to the project
 			final List<Contributor> contributors = projectPermissionsService.getContributors(rebacProject);
 			for (final Contributor contributor : contributors) {
 				if (contributor.isUser()) {
 					final RebacUser rebacUser = new RebacUser(contributor.getUserId(), reBACService);
-					rebacUser.removeAllRelationships(rebacProject);
+
+					// When we make a project a sample project,
+					// Update all user permissions to READER only
+					if (isSample) {
+						projectPermissionsService.removeProjectPermissions(
+							rebacProject,
+							rebacUser,
+							Schema.Relationship.CREATOR.toString()
+						);
+						projectPermissionsService.removeProjectPermissions(
+							rebacProject,
+							rebacUser,
+							Schema.Relationship.WRITER.toString()
+						);
+						projectPermissionsService.setProjectPermissions(
+							rebacProject,
+							rebacUser,
+							Schema.Relationship.READER.toString()
+						);
+					}
+
+					// When we revert a project to a non-sample project, and
+					// the user is the same as the project author it will become the creator of the project once more
+					if (!isSample && contributor.getUserId().equals(project.get().getUserId())) {
```

Comment:
This code block should never execute (to my knowledge) as the only contributor after a project is made as a `sample project` would be the ASKEM Admin group, therefore this shouldn't occur.  Instead I would suggest simple setting the project's userId to be the creator (assuming that is the original owner).

Reply:
FYI - I tested this with "ursula" and "adam" and once the project was de-sampled ursula was not made the creator.

Reply:
Good catch!
I changed the condition test to be clearer.